src/aircrack-ng.c

```diff
@@ -392,6 +396,211 @@ static void ac_aplist_free(void)
 	pthread_mutex_unlock(&mx_apl);
 }
 
+/**
+ * Release all unused AP base-stations stored in \a access_points, keeping
+ * only the specified \a ap_cur AP base-station.
+ *
+ * @param ap_cur The AP base-station to keep.
+ */
+static void ap_avl_release_unused(struct AP_info *ap_cur)
```

Comment:
I was thinking about this too and was going with a "mark and sweep" approach.
I think we have an `avl_tree` of "targets" that got created based on the marked ones from the entire list.
We should be able to free all of `access_points` once we've finished our two passes and only use "targets".

Both these ways to us to the same place. I think we could probably call `ac_list_free` once we populate targets, which is more or less what this does.